pkg/storage/formattedstore/formattedstore.go

```diff
@@ -122,14 +168,31 @@ func (s *formatStore) Get(k string) ([]byte, error) {
 		return nil, fmt.Errorf(failParseFormattedValue, err)
 	}
 
+	// put data in cache to retrieve it next time
+	if s.cacheStore != nil {
+		if err := s.cacheStore.Put(k, value); err != nil {
+			return nil, err
+		}
+	}
+
 	return value, nil
+
 }
 
 // The "don't filter" switch is here since it's possible for the underlying storage to have already done
 // the filtering, in which case it would be redundant to do it here too.
 // TODO (#2315) While doing the storage interface rework, see if the skipIteratorFiltering switch is still
 //  needed.
 func (s *formatStore) Iterator(startKey, endKey string) storage.StoreIterator {
+	if s.cacheStore != nil {
+		r := s.cacheStore.Iterator(startKey, endKey)
+		if r.Error() == nil {
+			return r
+		}
+
+		logger.Debugf("failed to get startKey %s endKey %s from cache store", startKey, endKey)
+	}
+
 	formattedIterator := s.store.Iterator(startKey, endKey)
```

Comment:
Need to use `r` down here if the cacheStore exists

Reply:
If the data is already formatted and you know the filtering has already been done, then you can disregard my comment

Reply:
this logic is doing that https://github.com/hyperledger/aries-framework-go/pull/2349/files#diff-8c872e6990542239384a04aec51b7f81b13913192843fad32b1f20f7c4afd06bR189